src/node_modules.cc

```diff
@@ -0,0 +1,442 @@
+              package_config.exports = std::string(exports);
+            }
+            break;
+          }
+          default:
+            break;
+        }
+      }
+    } else if (key == "imports") {
+      simdjson::ondemand::json_type imports_type;
+      if (!value.type().get(imports_type)) {
+        std::string_view imports;
+        switch (imports_type) {
+          case simdjson::ondemand::json_type::object: {
+            simdjson::ondemand::object imports_object;
+            if (!value.get_object().get(imports_object) &&
+                !imports_object.raw_json().get(imports)) {
+              package_config.imports = std::string(imports);
+            }
+            break;
+          }
+          case simdjson::ondemand::json_type::array: {
+            simdjson::ondemand::array imports_array;
+            if (!value.get_array().get(imports_array) &&
+                !imports_array.raw_json().get(imports)) {
+              package_config.imports = std::string(imports);
+            }
+            break;
+          }
+          case simdjson::ondemand::json_type::string: {
+            if (!value.get_string().get(imports)) {
+              package_config.imports = std::string(imports);
+            }
+            break;
+          }
+          default:
+            break;
+        }
+      }
+    } else if (key == "type") {
+      std::string_view type;
+      // Only update type if it is "commonjs" or "module".
+      if (!value.get_string().get(type) &&
+          (type == "commonjs" || type == "module")) {
+        package_config.type = std::string(type);
+      }
+    }
+  }
+
+  package_configs_.insert_or_assign(std::string(path.data(), path.size()),
+                                    package_config);
+
+  return package_config;
+}
+
+void BindingData::ReadPackageJSON(const FunctionCallbackInfo<Value>& args) {
+  CHECK_GE(args.Length(), 1);  // path, [is_esm, base, specifier]
+  CHECK(args[0]->IsString());  // path
+
+  Realm* realm = Realm::GetCurrent(args);
+  auto isolate = realm->isolate();
+
+  Utf8Value path(isolate, args[0]);
+  bool is_esm = args[1]->IsTrue();
+  auto error_context = ErrorContext();
+  if (is_esm) {
+    CHECK(args[2]->IsUndefined() || args[2]->IsString());  // base
+    CHECK(args[3]->IsString());                            // specifier
+
+    if (args[2]->IsString()) {
+      Utf8Value base_value(isolate, args[2]);
+      error_context.base = base_value.ToString();
+    }
+    Utf8Value specifier(isolate, args[3]);
+    error_context.specifier = specifier.ToString();
+  }
+
+  THROW_IF_INSUFFICIENT_PERMISSIONS(
+      realm->env(),
+      permission::PermissionScope::kFileSystemRead,
+      path.ToStringView());
+
+  auto package_json = GetPackageJSON(
+      realm, path.ToStringView(), is_esm ? &error_context : nullptr);
+  if (!package_json.has_value()) {
+    return;
+  }
+
+  args.GetReturnValue().Set(package_json->Serialize(realm));
+}
+
+void BindingData::GetClosestPackageJSONType(
```

Comment:
Does it make sense and generalize this to support passing the type we want to extract?

Reply:
Probably yes. I just dont know if we need multiple params yet. But good recommendation!

Reply:
Makes sense, I was thinking something like `getClosestPackageJSONProperty(property_str)`.